src/main/java/aeneas/models/Bullpen.java

```diff
@@ -4,41 +4,108 @@
 
 /**
  * A bullpen is a list of pieces available to the player.
- * 
+ *
  * @author Joseph Martin
  */
 public class Bullpen implements java.io.Serializable {
+
+  /**
+   * @return the logic
+   */
+  public BullpenLogic getLogic() {
+    return logic;
+  }
+
+  public static class BullpenLogic implements java.io.Serializable {
+    private boolean canReturnPiece;
+    private boolean canAddNewPiece;
+
+    public static BullpenLogic editorLogic() {
+      BullpenLogic b = new BullpenLogic();
+      b.canAddNewPiece = true;
+      b.canReturnPiece = true;
+      return b;
+    }
+
+    public static BullpenLogic puzzleLogic() {
+      BullpenLogic b = new BullpenLogic();
+      b.canAddNewPiece = false;
+      b.canReturnPiece = true;
+      return b;
+    }
+
+    public static BullpenLogic lightningLogic() {
+      BullpenLogic b = new BullpenLogic();
+      b.canAddNewPiece = false;
+      b.canReturnPiece = false;
+      return b;
+    }
+
+    public static BullpenLogic releaseLogic() {
+      BullpenLogic b = new BullpenLogic();
+      b.canAddNewPiece = false;
+      b.canReturnPiece = false;
+      return b;
+    }
+
+    /**
+     * @return the canReturnPiece
+     */
+    public boolean isCanReturnPiece() {
+      return canReturnPiece;
+    }
+
+    /**
+     * @return the canAddNewPiece
+     */
+    public boolean isCanAddNewPiece() {
+      return canAddNewPiece;
+    }
+  }
+
   ArrayList<Piece> pieces;
+  BullpenLogic logic;
 
   /**
    * Construct a bullpen with an empty list of pieces.
+   * @param logic the type of logic
    */
-  public Bullpen() {
+  public Bullpen(BullpenLogic logic) {
     this.pieces = new ArrayList<Piece>();
+    this.logic = logic;
   }
 
   /**
    * Construct a bullpen with the given list of pieces
    * @param pieces The initial list of pieces for the bullpen to contain.
+   * @param logic the type of logic
    */
-  public Bullpen(ArrayList<Piece> pieces) {
+  public Bullpen(BullpenLogic logic, ArrayList<Piece> pieces) {
     this.pieces = pieces;
+    this.logic = logic;
   }
 
   /**
    * Adds a piece to he bullpen
    * @param piece the piece to add
    */
-  void addPiece(Piece piece) {
-    pieces.add(piece);
+  public void addPiece(Piece piece) {
+    this.pieces.add(piece);
   }
 
   /**
    * Removes the given piece from the bullpen, if present.
    * @param piece the piece to remove.
    * @return true if piece was found and removed, false otherwise.
    */
-  boolean removePiece(Piece piece) {
-    return pieces.remove(piece);
+  public boolean removePiece(Piece piece) {
+    return this.pieces.remove(piece);
+  }
+
+  /**
+   * @return the pieces
+   */
+  public ArrayList<Piece> getPieces() {
```

Comment:
Good question--do we use this function? We're gonna need it at some point because a piece you drag needs to be removed